great_expectations/datasource/fluent/config.py

```diff
@@ -246,127 +254,16 @@ def yaml(
         intermediate_json_dict = self._exclude_name_fields_from_fluent_datasources(
             config=intermediate_json_dict
         )
+        deep_filter_properties_iterable(
+            properties=intermediate_json_dict, clean_falsy=True, inplace=True
```

Comment:
Why are we doing this?
This seems wrong and feels like a hidden foot gun waiting to go off.

Reply:
@Kilo59 Indeed, we were asked to not include `assets` in the `YAML` representation, if there are no data assets in the given `Datasource` configuration, rather than to show an empty dictionary in the `YAML` string.  I am not sure if this is documented, but I distinctly remember being asked to do that (otherwise, I would not have embarked on this step).

So your assumption was correct.  The only times that using `deep_filter_properties_iterable()` is not appropriate is if there are possibilities of having keys in `YAML` with `NULL` (or empty) values, whereby the keys must still be shown.  If this will not be the case with the `Fluent` Datasources, then we can keep using `deep_filter_properties_iterable()`.  However, if you feel that there will be such cases potentially in the future, then `deep_filter_properties_iterable()` will not work.  Could you please describe the approach you had in mind for solving it?  Is it looping through the configuration of every `Datasource` and removing only the `assets` key (if it has a `Falsy`) value?  Please let me know so that we can finalize the implementation.  Thank you.

Reply:
I think the point @Kilo59 is making is that we can handle this logic via `pydantic` with the overloaded `yaml()` method, like we did to remove `name`.

Reply:
@NathanFarmer I would be curious to see the description (and happy to implement it) -- I handled the `name` using the validator, though the overloaded methods differ only in their method signature; is this not the case?  Thanks. /cc @Kilo59

Reply:
Here's a draft PR for the `assets` key fix.
https://github.com/great-expectations/great_expectations/pull/7341